#!/usr/bin/env python
PACKAGE = "feature_extraction"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kinect_delay", double_t, 0, "kinect delay", 0.0, -1.0, 1.0)
gen.add("depth_scale", double_t, 0, "kinect depth scale", 1.0, 0.0, 2.0)
gen.add("number_of_features", int_t, 0, "number of features to extract", 300, 50, 400)
gen.add("feature_max_depth", double_t, 0, "maximum depth for valid 3d features", 3., 0., 100.)
gen.add("use_bilateral_filter", bool_t, 0, "true, if bilateral filter should be applied to the depth images", True)
gen.add("store_depth_images", bool_t, 0, "true, if the depth image message should be added to the node data", True)
gen.add("store_color_images", bool_t, 0, "true, if the color image message should be added to the node data", True)
gen.add("store_laser_scans", bool_t, 0, "true, if a laser scan should be extracted from the depth image and added to the node data", True)

exit(gen.generate(PACKAGE, "feature_extraction", "FeatureExtraction"))
